contrib/guix/libexec/prelude.bash

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+export LC_ALL=C
+set -e -o pipefail
+
+# shellcheck source=contrib/shell/realpath.bash
+source contrib/shell/realpath.bash
+
+# shellcheck source=contrib/shell/git-utils.bash
+source contrib/shell/git-utils.bash
+
+################
+# Required non-builtin commands should be invocable
+################
+
+check_tools() {
+    for cmd in "$@"; do
+        if ! command -v "$cmd" > /dev/null 2>&1; then
+            echo "ERR: This script requires that '$cmd' is installed and available in your \$PATH"
+            exit 1
+        fi
+    done
+}
+
+check_tools cat env readlink dirname basename git
+
+################
+# We should be at the top directory of the repository
+################
+
+same_dir() {
+    local resolved1 resolved2
+    resolved1="$(bash_realpath "${1}")"
```

Comment:
Does `realpath -e` not work on these machines or something?

Reply:
I don't we actually need a portable implementation here, since `realpath` is available by default on practically every Linux distribution.